zerver/lib/actions.py

```diff
@@ -5412,11 +5412,12 @@ def do_add_default_stream(stream: Stream) -> None:
         notify_default_streams(stream.realm)
 
 
+@transaction.atomic
 def do_remove_default_stream(stream: Stream) -> None:
```

Comment:
do we consider these to deserve their own RealmAuditLogs?

Reply:
Probably; though there's also an argument that we should rework this into being a flag on `Stream` rather than a separate table, which would then suggest the opposite decision.

Reply:
Okay - in that case, does it makes sense to leave this commit as it is, just adding the `atomic` part and not doing anything about RealmAuditLog for now, leaving that matter for the future decision?

Reply:
I think that's OK -- we do want to end our sweep with feeling good about the state of these details, but it's fine to defer this one until we are further in product design.